daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/Ref.scala

```diff
@@ -285,4 +285,32 @@ object Ref {
   type TypeSynName = Identifier
   val TypeSynName = Identifier
 
+  /*
+     max size when converted to string: 3068 ASCII chars
+   */
+  final case class QualifiedChoiceName(interfaceId: Option[Identifier], choiceName: ChoiceName) {
+    override def toString: String = interfaceId match {
+      case None => choiceName
+      case Some(ifaceId) => "#" + ifaceId.toString + "#" + choiceName
```

Comment:
Nit pick: is there any value in using string interpolation here? Not particularly fused in this case, simply pointing this out.